### PR TITLE
feat: hardcode grafbase otel endpoint

### PR DESCRIPTION
**Pull request by Finistere**
I'm hardcoding our OTEL endpoint whenever a graph reference is used. We don't fail if the token isn't present because the gateway will fail later with a better error message anyway. Using native certs ended up being as easy as just adding the right feature flag in `tonic`...

### CODE REVIEW

File: gateway/crates/federated-server/src/server/graph_fetch_method.rs [hunk 1]
```diff
@@ -37,12 +36,12 @@ impl GraphFetchMethod {
         sender: watch::Sender<Option<Arc<Engine>>>,
     ) -> crate::Result<()> {
         match self {
-            #[cfg(not(feature = "lambda"))]
             GraphFetchMethod::FromApi {
                 access_token,
                 graph_name,
                 branch,
             } => {
+                #[cfg(not(feature = "lambda"))]
```
Comment (pimeys) [inline gateway/crates/federated-server/src/server/graph_fetch_method.rs:44]: ?

File: gateway/crates/gateway-binary/src/args.rs [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
+    fn fetch_method(&self) -> anyhow::Result<GraphFetchMethod>;
+    fn config(&self) -> anyhow::Result<Config>;
+    fn log_format<S>(&self) -> BoxedLayer<S>
+    where
+        S: Subscriber + for<'span> LookupSpan<'span> + Send + Sync;
+}
 
-#[cfg(not(feature = "lambda"))]
-pub(crate) use self::std::Args;
+pub(crate) fn parse() -> impl Args {
```
Comment (pimeys) [inline gateway/crates/gateway-binary/src/args.rs:24]: :3

File: gateway/crates/gateway-binary/src/args/std.rs [hunk 3]
```diff
@@ -75,17 +78,38 @@ impl Args {
...
+        let mut config = match self.config.as_ref() {
             Some(path) => {
                 let config = fs::read_to_string(path).context("could not read config file")?;
-                Ok(toml::from_str(&config)?)
+                toml::from_str(&config)?
             }
-            None => Ok(Config::default()),
+            None => Config::default(),
+        };
+        if let Some(ref token) = self.grafbase_access_token {
```
Comment (pimeys) [inline gateway/crates/gateway-binary/src/args/std.rs:89]: So we still need some way of doing this in AWS Lambda?

**Review by pimeys — APPROVED** [review]

File: gateway/crates/federated-server/src/server/graph_fetch_method.rs (hunk 1 above)
Comment (Finistere) [reply gateway/crates/federated-server/src/server/graph_fetch_method.rs:44]: The code is still behind the `not(lambda)` flag

File: gateway/crates/gateway-binary/src/args/std.rs (hunk 3 above)
Comment (Finistere) [reply gateway/crates/gateway-binary/src/args/std.rs:89]: Well yes, not sure how.

File: gateway/crates/gateway-binary/src/args.rs (hunk 2 above)
Comment (Finistere) [reply gateway/crates/gateway-binary/src/args.rs:24]: didn't want to write anymore code hidden behind a `not(feature = "lambda")` flag 😅

**Pull request merged**